server/src/Irony.cpp

```diff
@@ -326,4 +327,34 @@ void Irony::complete(const std::string &file,
     clang_disposeCodeCompleteResults(completions);
     std::cout << ")\n";
   }
+
+}
+
+// Get compile options from JSON database
+void Irony::getCompileOptions(const std::string &projectRoot,
+                              const std::string &file) {
+  std::vector<std::vector<std::string>> flags = getFlags(projectRoot, file);
+
+  if (!flags.empty()) {
+    std::cout << "(\n";
+
+    for (auto vi = flags.begin(), vend = flags.end(); vi != vend; ++vi) {
+      std::cout << "( ";
+
+      std::cout << "(";
+      for (auto it = std::next(vi->begin()), end = vi->end(); it != end; ++it)
```

Comment:
The build-directory being first element of flags is weird, we may be better off with a pair.